pkg/datapath/loader/base.go

```diff
@@ -239,6 +236,15 @@ func (l *Loader) ReinitializeXDP(ctx context.Context, extraCArgs []string) error
 	return nil
 }
 
+// ReinitializeXDP (re-)configures the XDP datapath only. This includes recompilation
+// and reinsertion of the object into the kernel as well as an atomic program replacement
+// at the XDP hook. extraCArgs can be passed-in in order to alter BPF code defines.
+func (l *Loader) ReinitializeXDP(ctx context.Context, o datapath.BaseProgramOwner, extraCArgs []string) error {
+	o.GetCompilationLock().Lock()
```

Comment:
Instead of using `o.GetCompilationLock().Lock()`, couldn't we have introduced `Loader.XDPLock` instead? This would make it less error prone, as we would know that always the same lock is taken vs the current approach where we pass a lock owner which we can easily mix resulting in false protection.

Also, it would be good to have a code comment saying why the lock is needed.

Reply:
> This would make it less error prone, as we would know that always the same lock is taken vs the current approach where we pass a lock owner which we can easily mix resulting in false protection.

Hm, the lock owner in this case is actually always the daemon itself upon call to Reinitialize(). It would actually be fatal if it wasn't. I'm not sure why we have `o datapath.BaseProgramOwner` in there, might as well be a concrete lock object that is tied to the daemon only.